src/compute/src/sink/persist_sink.rs

```diff
@@ -136,12 +136,14 @@ where
             let mut stash: HashMap<_, Vec<_>> = HashMap::new();
 
             // TODO(aljoscha): We need to figure out what to do with error results from these calls.
-            let mut write = persist_clients
+            let client = persist_clients
                 .lock()
                 .await
                 .open(persist_location)
                 .await
-                .expect("could not open persist client")
+                .expect("could not open persist client");
```

Comment:
Splitting this method chain into two makes it so the `MutexGuard` from `persist_clients.lock()` can already be dropped here, and isn't held across the `open_writer` await. Without this change creation of the logging sinks deadlocks.

Other places that use `persist_clients` behind a mutex also use this pattern, so this is probably a good change to make even if we decide against merging this PR.